src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
+                                       (map :?o r))))]
 
     ;; Should not match when negation is met.
-    (is (empty? (-> s
-                    (insert (->Temperature 10 "MCI")
-                            (->Cold 10))
-                    (fire-rules)
-                    (query cold-not-match-temp))))
+    (let [end-session (-> s
+                          (insert (->Temperature 10 "MCI")
+                                  (->Cold 10))
+                          (fire-rules))]
+      (is (empty? (query end-session cold-not-match-temp)))
+      (is (no-system-types? end-session)))
 
     ;; Should have result if only a single item matched.
-    (is (= [{}]
-           (-> s
-               (insert (->Temperature 10 "MCI"))
-               (fire-rules)
-               (query cold-not-match-temp))))
+    (let [end-session (-> s
+                          (insert (->Temperature 10 "MCI"))
+                          (fire-rules))]
+      (is (= [{}]
+             (query end-session cold-not-match-temp)))
+      (is (no-system-types? end-session)))
 
     ;; Test previous binding is visible.
-    (is (empty? (-> s-with-prior
-                    (fire-rules)
-                    (query negation-with-prior-bindings))))
+    (let [end-session (fire-rules s-with-prior)]
+      (is (empty? (-> s-with-prior
+                      (fire-rules)
+                      (query negation-with-prior-bindings))))
+      (is (no-system-types? end-session)))
 
     ;; Should have result since negation does not match.
-    (is (= [{:?l "MCI"}]
-           (-> s-with-prior
-               (insert (->WindSpeed 10 "MCI")
-                       (->Temperature 10 "ORD")
-                       (->Cold 10))
-               (fire-rules)
-               (query negation-with-prior-bindings))))
+    (let [end-session (-> s-with-prior
+                          (insert (->WindSpeed 10 "MCI")
+                                  (->Temperature 10 "ORD")
+                                  (->Cold 10))
+                          (fire-rules))]
+      (is (= [{:?l "MCI"}]
+             (query end-session negation-with-prior-bindings)))
+      (is (no-system-types? end-session)))
 
     ;; No result because negation matches.
-    (is (empty? (-> s-with-prior
-                    (insert (->WindSpeed 10 "MCI")
-                            (->Temperature 10 "MCI")
-                            (->Cold 10))
-                    (fire-rules)
-                    (query negation-with-prior-bindings))))
+    (let [end-session (-> s-with-prior
+                          (insert (->WindSpeed 10 "MCI")
+                                  (->Temperature 10 "MCI")
+                                  (->Cold 10))
+                          (fire-rules))]
+      (is (empty? (query end-session negation-with-prior-bindings)))
+      (is (no-system-types? end-session)))
 
     ;; There should be only one root to the beta tree because the top condition is reused.
```

Comment:
I don't know, I was just setting it aside for now since it was commented out already.  It seems undesirable to write tests directly against to-beta-tree though, not sure why it was done in the first place.  It can probably just be deleted.